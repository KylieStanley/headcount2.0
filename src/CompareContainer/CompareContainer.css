h3 {
  margin: 10px 0;
  text-align: center;
}

.compare-container {
  display: flex;
  justify-content: center;
  align-items: center;
}