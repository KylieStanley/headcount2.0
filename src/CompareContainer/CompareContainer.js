import React from 'react';
import Card from '../Card/Card'
import CompareCard from '../CompareCard/CompareCard'
import './CompareContainer.css'
import PropTypes from 'prop-types';


const CompareContainer = ({ selectedCards, compareDistrictAverages, selectCard }) => {

  let selected = selectedCards.map((district, i) => {
                  return <Card district={district} key={i} selectedCards={selectedCards} selectCard={selectCard}/>
                 })

  if (selectedCards.length === 2) {
    var district1 = selectedCards[0].location;
    var district2 = selectedCards[1].location;

    var comparedDistricts = compareDistrictAverages(district1, district2);
  }

  selected.splice(1, 0, <CompareCard selectedCards={selectedCards} district1={district1} 
                                     district2={district2} comparedDistricts={comparedDistricts} key={3}/>)

  return (
    <div className="compare-container">
      { selected }
    </div>
  );
}

CompareContainer.propTypes = {
  selectedCards: PropTypes.array.isRequired,
  compareDistrictAverages: PropTypes.func.isRequired
}

export default CompareContainer;
