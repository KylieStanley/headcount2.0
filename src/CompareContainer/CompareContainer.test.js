import React from 'react';
import CompareContainer from './CompareContainer.js';
import {shallow, mount} from 'enzyme';

describe('CompareContainer', () => {
  it('should match the snapshot with data passed in correctly', () => {
    const wrapper = shallow( <CompareContainer 
      selectedCards={[{location: "COLORADO SPRINGS",
                       stats: {2005: .4}
                      }]}
      compareDistrictAverages={jest.fn()}
    />)

    expect(wrapper).toMatchSnapshot();
  })
})