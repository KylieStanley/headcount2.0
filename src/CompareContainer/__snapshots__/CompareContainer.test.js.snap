// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareContainer should match the snapshot with data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareContainer
    compareDistrictAverages={[Function]}
    selectedCards={
        Array [
            Object {
              "location": "COLORADO SPRINGS",
              "stats": Object {
                "2005": 0.4,
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          district={
                    Object {
                              "location": "COLORADO SPRINGS",
                              "stats": Object {
                                "2005": 0.4,
                              },
                            }
          }
          selectCard={undefined}
          selectedCards={
                    Array [
                              Object {
                                "location": "COLORADO SPRINGS",
                                "stats": Object {
                                  "2005": 0.4,
                                },
                              },
                            ]
          }
/>,
        <CompareCard
          comparedDistricts={undefined}
          district1={undefined}
          district2={undefined}
          selectedCards={
                    Array [
                              Object {
                                "location": "COLORADO SPRINGS",
                                "stats": Object {
                                  "2005": 0.4,
                                },
                              },
                            ]
          }
/>,
        <BlankCard />,
      ],
      "className": "compare-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "COLORADO SPRINGS",
        "nodeType": "function",
        "props": Object {
          "district": Object {
            "location": "COLORADO SPRINGS",
            "stats": Object {
              "2005": 0.4,
            },
          },
          "selectCard": undefined,
          "selectedCards": Array [
            Object {
              "location": "COLORADO SPRINGS",
              "stats": Object {
                "2005": 0.4,
              },
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "comparedDistricts": undefined,
          "district1": undefined,
          "district2": undefined,
          "selectedCards": Array [
            Object {
              "location": "COLORADO SPRINGS",
              "stats": Object {
                "2005": 0.4,
              },
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "card2",
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            district={
                        Object {
                                    "location": "COLORADO SPRINGS",
                                    "stats": Object {
                                      "2005": 0.4,
                                    },
                                  }
            }
            selectCard={undefined}
            selectedCards={
                        Array [
                                    Object {
                                      "location": "COLORADO SPRINGS",
                                      "stats": Object {
                                        "2005": 0.4,
                                      },
                                    },
                                  ]
            }
/>,
          <CompareCard
            comparedDistricts={undefined}
            district1={undefined}
            district2={undefined}
            selectedCards={
                        Array [
                                    Object {
                                      "location": "COLORADO SPRINGS",
                                      "stats": Object {
                                        "2005": 0.4,
                                      },
                                    },
                                  ]
            }
/>,
          <BlankCard />,
        ],
        "className": "compare-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "COLORADO SPRINGS",
          "nodeType": "function",
          "props": Object {
            "district": Object {
              "location": "COLORADO SPRINGS",
              "stats": Object {
                "2005": 0.4,
              },
            },
            "selectCard": undefined,
            "selectedCards": Array [
              Object {
                "location": "COLORADO SPRINGS",
                "stats": Object {
                  "2005": 0.4,
                },
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "comparedDistricts": undefined,
            "district1": undefined,
            "district2": undefined,
            "selectedCards": Array [
              Object {
                "location": "COLORADO SPRINGS",
                "stats": Object {
                  "2005": 0.4,
                },
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "card2",
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
